addons/folo/common/src/main/java/org/commonjava/indy/folo/change/FoloTrackingListener.java

```diff
@@ -114,6 +114,11 @@ public void onFileAccess( @Observes final FileAccessEvent event )
             logger.trace( "Tracking report: {} += {} in {} (DOWNLOAD)", trackingKey, transfer.getPath(),
                           keyedLocation.getKey() );
 
+            //FIXME: Here we need to think about npm metadata retrieving case. As almost all npm metadata retrieving is through
```

Comment:
Do we know that this will cause problems currently? I guess the main problem is that it doesn't match exactly with the actual requests that were made, and it might be a bit harder to know what NPM packages were requested. OTOH, if we don't have a matching use case on the client side yet, and we can predictably extract a package name from these records, we should be okay I think.

Reply:
I'm not sure if there is case that will do listing on root path. I agree to remove this. Will do.

Reply:
Because there will be 2 paths in tracking record: original and local. The original one is to track the remote, which the real path is /$pkg but changed to /$path/package.json in our case. I agree that the local one can be the package.json one, but I'm not sure if original one should be. So that's why I'm adding this FIXME here. I'm not sure if this original will be used in any cases, because if it will, that path is wrong here. I will leave the concern here as a comment with question but not FIXME.